### PR TITLE
Sync V2 WebUI [base on sync-v2]

### PR DESCRIPTION
**Pull request by petemill**
This is the style, functionality and updated strings for Sync V2 (in the Settings WebUI).

...everything except QR code generation...

image

image

image

image

image

image

image

image

image

### CODE REVIEW

File: browser/resources/settings/brave_sync_page/brave_sync_configure.html [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+      }
+
+    </style>
+
+    <!-- Device list goes here -->
+    <section class="devices">
+      <h1>Device List</h1>
+      <p class="subtitle">
+        The device list below includes all devices in your sync chain.
+        Each device can be managed from any other device.
```
Comment (petemill) [inline browser/resources/settings/brave_sync_page/brave_sync_configure.html:89]: Needs i18n

File: browser/resources/settings/brave_sync_page/brave_sync_configure.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    this.syncCodeDialogType_ = 'choose'
+  },
+
+  onSyncCodeDialogDone_: function() {
+    this.syncCodeDialogType_ = null
+  },
+
+  onResetSyncChain_: async function() {
+    // TODO(petemill): translate message
+    const messageText = 'If you reset Sync, you will have to re-enter a sync code from another device in order to sync with it. Are you sure you want to proceed?'
```
Comment (petemill) [inline browser/resources/settings/brave_sync_page/brave_sync_configure.js:85]: Need i18n

**Comment by bradleyrichter** [comment]
@petemill all of the UI looks pretty tight. One area to refine is the device list. See what you think of this version that removes HRs and adds some space:

image

**Comment by bsclifton** [comment]
Build is failing for me on Windows:
```
[16233/43035] ACTION //chrome/browser/resources/settings:build(//build/toolchain/win:win_clang_x64)
FAILED: gen/chrome/browser/resources/settings/vulcanized.html gen/chrome/browser/resources/settings/lazy_load.vulcanized.html gen/chrome/browser/resources/settings/crisper.js gen/chrome/browser/resources/settings/lazy_load.crisper.js
C:/bb1/vendor/depot_tools/bootstrap-3_8_0_chromium_8_bin/python/bin/python.exe ../../chrome/browser/resources/optimize_webui.py --host settings --input gen/chrome/browser/resources/settings/settings_resources.unpak --out_folder gen/chrome/browser/resources/settings --depfile gen/chrome/browser/resources/settings/build.d --js_out_files crisper.js lazy_load.crisper.js --exclude chrome://resources/mojo/chromeos/services/network_config/public/mojom/cros_network_config.mojom.html --html_in_files settings.html lazy_load.html --html_out_files vulcanized.html lazy_load.vulcanized.html --insert_in_head "<base href=\"chrome://settings\">"
Traceback (most recent call last):
  File "../../chrome/browser/resources/optimize_webui.py", line 399, in <module>
    main(sys.argv[1:])
  File "../../chrome/browser/resources/optimize_webui.py", line 393, in main
    '\n'.join(manifest['_missing']))
Exception: polymer-bundler could not find files for the following URLs:
brave_sync_page/brave_sync_control.html
```

File: browser/ui/webui/settings/brave_sync_handler.cc [hunk 3]
```diff
@@ -99,7 +115,17 @@ void BraveSyncHandler::HandleGetQRCode(const base::ListValue* args) {
...
-  // TODO(petemill): Wait until at least cr82 for QRCodeGenerator to be in tree.
+
+  std::vector<uint8_t> seed;
+  if (!brave_sync::crypto::PassphraseToBytes32(sync_code->GetString(), &seed)) {
+    LOG(ERROR) << "invalid sync code:" << sync_code->GetString();
+    ResolveJavascriptCallback(*callback_id, base::Value(false));
+    return;
+  }
+
+  std::string seed_serialized = StrFromUint8Array(seed);
```
Comment (darkdh) [inline browser/ui/webui/settings/brave_sync_handler.cc:126]: Why do we need this? This is not required for `QRCodeGenerator::Generate`

Comment (petemill) [reply browser/ui/webui/settings/brave_sync_handler.cc:126]: I will need `StrFromUint8Array` to send the output of QRCode::Generate, if I'm rendering it in JS 😞

File: browser/ui/webui/settings/brave_sync_handler.cc [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+      "SyncGetQRCode",
+      base::BindRepeating(&BraveSyncHandler::HandleGetQRCode,
+                          base::Unretained(this)));
+  web_ui()->RegisterMessageCallback(
+      "SyncSetupReset", base::BindRepeating(&BraveSyncHandler::HandleReset,
+                                            base::Unretained(this)));
+}
+
+void BraveSyncHandler::OnJavascriptAllowed() {
+   syncer::DeviceInfoTracker* tracker =
```
Comment (darkdh) [inline browser/ui/webui/settings/brave_sync_handler.cc:67]:
```
browser/ui/webui/settings/brave_sync_handler.cc:67:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
browser/ui/webui/settings/brave_sync_handler.cc:212:  Add #include <utility> for move  [build/include_what_you_use] [4]
browser/ui/webui/settings/brave_sync_handler.cc:156:  Add #include <vector> for vector<>  [build/include_what_you_use] [4]
```
lint errors

Comment (petemill) [reply browser/ui/webui/settings/brave_sync_handler.cc]: fixed, thanks for noticing that 😥

**Comment by petemill** [comment]
@bradleyrichter, updated devicelist table:

image

**Pull request merged**